### PR TITLE
Make beaker runnable without spark + create more examples.

### PR DESCRIPTION
**Pull request by nishant-deshpande-db**
Make beaker runnable without spark + create more examples.
Allow semicolon delimited sql file input + a directory of sql files. 
Create more examples and fix bugs to make them runnable.

### CODE REVIEW

**Comment by nishant-deshpande-db** [comment]
I realize I also changed the way the queries get executed. Sorry I didn't put this in the comments - I will add this. I did it a while back.

I create a single threadpool with n threads (n = concurrency) and then execute all queries using this threadpool.

Vs creating different buckets of n queries and executing the queries in the bucket concurrently.

File: src/beaker/benchmark.py [hunk 1]
```diff
@@ -120,18 +130,30 @@ def _execute_single_query(self, query, id=None):
...
         elapsed_time = f"{end_time - start_time:0.3f}"
-        metrics = [(id, self.hostname, self.http_path, self.concurrency, query, elapsed_time)]
-        metrics_df = self.spark.createDataFrame(metrics,
-                                                "id string, hostname string, warehouse string, concurrency int, query_text string, query_duration_secs string")
-        return metrics_df
+        #metrics = (id, self.hostname, self.http_path, self.concurrency, query, elapsed_time)
+        metrics = {'id':id, 'hostname':self.hostname, 'http_path':self.http_path,
+                   'concurrency':self.concurrency, 'query':query,
+                   'elapsed_time':elapsed_time}
+        #metrics_df = self.spark.createDataFrame(metrics,
```
Comment (mberk06) [inline src/beaker/benchmark.py:143]: Can we clean up these comments?

File: src/beaker/benchmark.py [hunk 2]
```diff
@@ -120,18 +130,30 @@ def _execute_single_query(self, query, id=None):
...
+            self.hostname, self.http_path, self.token, self.catalog, self.schema,
+            self.results_cache_enabled)
         sql_warehouse.execute_query(query)
         end_time = time.perf_counter()
         elapsed_time = f"{end_time - start_time:0.3f}"
-        metrics = [(id, self.hostname, self.http_path, self.concurrency, query, elapsed_time)]
-        metrics_df = self.spark.createDataFrame(metrics,
-                                                "id string, hostname string, warehouse string, concurrency int, query_text string, query_duration_secs string")
-        return metrics_df
+        #metrics = (id, self.hostname, self.http_path, self.concurrency, query, elapsed_time)
```
Comment (mberk06) [inline src/beaker/benchmark.py:139]: remove

File: src/beaker/benchmark.py [hunk 3]
```diff
@@ -140,84 +162,97 @@ def _parse_queries(self, raw_queries):
...
+        return metrics
+
+
+    def _execute_queries(self, queries, num_threads):
+        metrics_list = None
+        with ThreadPoolExecutor(max_workers=num_threads) as executor:
+            metrics_list = list(
+                executor.map(lambda x: self._execute_single_query(*x),
+                             queries))
+        #final_metrics_df = reduce(DataFrame.unionAll, metrics_list)
```
Comment (mberk06) [inline src/beaker/benchmark.py:232]: remove

File: src/beaker/sqlwarehouseutils.py [hunk 4]
```diff
@@ -131,5 +139,8 @@ def launch_warehouse(self, config):
                 }
             }
         )
-        warehouse_id = response.json()['id']
+        #logging.info(f"create cluster response: {response.json()}")
```
Comment (mberk06) [inline src/beaker/sqlwarehouseutils.py:142]: remove

File: src/beaker/benchmark.py (hunk 1 above)
Comment (nishant-deshpande-db) [reply src/beaker/benchmark.py:143]: Done

File: src/beaker/benchmark.py (hunk 2 above)
Comment (nishant-deshpande-db) [reply src/beaker/benchmark.py:139]: Done

File: src/beaker/benchmark.py (hunk 3 above)
Comment (nishant-deshpande-db) [reply src/beaker/benchmark.py:232]: Done

File: src/beaker/sqlwarehouseutils.py (hunk 4 above)
Comment (nishant-deshpande-db) [reply src/beaker/sqlwarehouseutils.py:142]: Done

**Review by mberk06 — APPROVED** [review]

**Pull request merged**